Sources/HttpParser.swift

```diff
@@ -75,10 +79,51 @@ public class HttpParser {
 //        }
     }
     
+    public func readContent(_ socket: Socket, request: HttpRequest, filePreprocess: Bool) throws {
+        guard let contentType = request.headers["content-type"],
+            let contentLength = request.headers["content-length"],
+            let contentLengthValue = Int(contentLength) else {
+            return
+        }
+        let isFileUpload = contentType == "application/octet-stream"
+        if isFileUpload && filePreprocess {
```

Comment:
Instead of or in addition to `filePreprocess` the data should go to disk is it's pretty big, so that the server doesn't run into memory exhaustion issues. The written file should be handed back to the caller instead of just closed, and it should be memory-mapped so that it can be efficiently processed by the caller.